kong/clustering/utils.lua

```diff
@@ -320,6 +324,25 @@ function _M.check_configuration_compatibility(obj, dp_plugin_map)
   return true, nil, CLUSTERING_SYNC_STATUS.NORMAL
 end
 
+local function parse_proxy_url(conf)
+  local ret = {}
+  local proxy_server = conf.proxy_server
+  if proxy_server then
+    -- assume proxy_server is validated in conf_loader
+    local parsed = parse_url(proxy_server)
+    ret.proxy_url = fmt("%s://%s:%s", parsed.scheme, parsed.host, parsed.port or 80)
```

Comment:
This won't work with HTTPS support. Just a note.

Reply:
good catch, move default to 443